lib/src/test/java/com/dynatrace/metric/util/NormalizeTest.java

```diff
@@ -177,12 +201,12 @@ private static String createNonsenseStringOfLength(int n) {
         Arguments.of("valid uppercase", "VALUE", "VALUE"),
         Arguments.of("valid colon", "a:3", "a:3"),
         Arguments.of("valid value 2", "~@#ä", "~@#ä"),
-        Arguments.of("escape spaces", "a b", "a\\ b"),
-        Arguments.of("escape comma", "a,b", "a\\,b"),
-        Arguments.of("escape equals", "a=b", "a\\=b"),
-        Arguments.of("escape backslash", "a\\b", "a\\\\b"),
-        Arguments.of("escape multiple invalids", " ,=\\", "\\ \\,\\=\\\\"),
-        Arguments.of("escape key-value pair", "key=\"value\"", "key\\=\"value\""),
+        Arguments.of("escape spaces", "a b", "a b"),
```

Comment:
are we not escaping spaces anymore?

Reply:
`dimensionValue()` only replaces invalid characters now (necessary for consistent deduplication) but the protocol-level escaping only happens on `serialize()`.
It would be better to rename the test cases to "valid space" and so on, since those should be kept as-is at this level.

Reply:
We have identified a problem that occurred when using the Metric builder. In some cases, dimension values would be escaped twice or even three times, e. g.:
the dimension value to escape: 	`"dimension value"`
sometimes we would get:			`"dimension\\\ value"`
and sometimes even:				`"dimension\\\\\\ value"`
because the previously escaped spaces lead to escaped backslashes, which themselves were escaped again, so basically backslashes all the way down. In order to mitigate that, I moved the escaping to a separate function that is called once in the `serialize` function, just before the export.